include/tvm/node/container.h

```diff
@@ -43,51 +44,1188 @@ using runtime::Downcast;
+    if (slots <= SmallMapNode::kMaxSize) {       \
+      TSmall var = static_cast<TSmall>(base);    \
+      body;                                      \
+    } else {                                     \
+      TDense var = static_cast<TDense>(base);    \
+      body;                                      \
+    }                                            \
+  }
+
+inline MapNode::iterator::pointer MapNode::iterator::operator->() const {
+  _TVM_DISPATCH_MAP_CONST(self, p, { return p->DeRefItr(i); });
+}
+
+inline MapNode::iterator& MapNode::iterator::operator++() {
+  _TVM_DISPATCH_MAP_CONST(self, p, {
+    i = p->IncItr(i);
+    return *this;
+  });
+}
+
+inline MapNode::iterator& MapNode::iterator::operator--() {
+  _TVM_DISPATCH_MAP_CONST(self, p, {
+    i = p->IncItr(i);
+    return *this;
+  });
+}
+
+inline size_t MapNode::count(const key_type& key) const {
+  _TVM_DISPATCH_MAP_CONST(this, p, { return p->count(key); });
+}
+
+inline const MapNode::mapped_type& MapNode::at(const MapNode::key_type& key) const {
+  _TVM_DISPATCH_MAP_CONST(this, p, { return p->at(key); });
+}
+
+inline MapNode::mapped_type& MapNode::at(const MapNode::key_type& key) {
+  _TVM_DISPATCH_MAP(this, p, { return p->at(key); });
+}
+
+inline MapNode::iterator MapNode::begin() const {
+  _TVM_DISPATCH_MAP_CONST(this, p, { return p->begin(); });
+}
+
+inline MapNode::iterator MapNode::end() const {
+  _TVM_DISPATCH_MAP_CONST(this, p, { return p->end(); });
+}
+
+inline MapNode::iterator MapNode::find(const MapNode::key_type& key) const {
+  _TVM_DISPATCH_MAP_CONST(this, p, { return p->find(key); });
+}
+
+inline void MapNode::erase(const MapNode::iterator& position) {
+  _TVM_DISPATCH_MAP(this, p, { return p->erase(position); });
+}
+
+#undef _TVM_DISPATCH_MAP
+#undef _TVM_DISPATCH_MAP_CONST
+
+inline ObjectPtr<MapNode> MapNode::Empty() { return SmallMapNode::Empty(); }
+
+inline ObjectPtr<MapNode> MapNode::CopyFrom(MapNode* m) {
+  if (m->slots_ <= SmallMapNode::kMaxSize) {
+    return SmallMapNode::CopyFrom(static_cast<SmallMapNode*>(m));
+  } else {
+    return DenseMapNode::CopyFrom(static_cast<DenseMapNode*>(m));
+  }
+}
+
+template <typename IterType>
+inline ObjectPtr<Object> MapNode::CreateFromRange(IterType first, IterType last) {
+  int64_t _cap = std::distance(first, last);
+  if (_cap < 0) {
+    return SmallMapNode::Empty();
+  }
+  uint64_t cap = static_cast<uint64_t>(_cap);
+  if (cap < SmallMapNode::kMaxSize) {
+    return SmallMapNode::CreateFromRange(cap, first, last);
+  }
+  uint32_t fib_shift;
+  uint64_t n_slots;
+  DenseMapNode::CalcTableSize(cap, &fib_shift, &n_slots);
+  ObjectPtr<Object> n = DenseMapNode::Empty(fib_shift, n_slots);
+  for (; first != last; ++first) {
+    KVType kv(*first);
+    DenseMapNode::InsertMaybeReHash(kv, &n);
+  }
+  return n;
+}
+
+inline void MapNode::InsertMaybeReHash(const KVType& kv, ObjectPtr<Object>* map) {
+  constexpr uint64_t kMaxSize = SmallMapNode::kMaxSize;
```

Comment:
kMaxSize -> kSmallMapMaxSize(needed to avoid constexpr symbol problem)

Reply:
Uh I was thinking the name can shadow SmallMapNode::kMaxSize symbol...Okay will do!